src/hsl.rs

```diff
@@ -36,60 +40,70 @@ pub struct Hsl<Wp = D65, T = f32>
     ///a clear color, and 1.0 will give white.
     pub lightness: T,
 
-    ///The white point associated with the color's illuminant and observer.
-    ///D65 for 2 degree observer is used by default.
-    pub white_point: PhantomData<Wp>,
+    ///The white point and RGB primaries this color is adapted to. The default
+    ///is the sRGB standard.
+    pub space: PhantomData<S>,
 
 }
 
-impl<Wp, T> Copy for Hsl<Wp, T>
+impl<S, T> Copy for Hsl<S, T>
     where T: Float,
-        Wp: WhitePoint<T>
+        S: RgbSpace
 {}
 
-impl<Wp, T> Clone for Hsl<Wp, T>
+impl<S, T> Clone for Hsl<S, T>
     where T: Float,
-        Wp: WhitePoint<T>
+        S: RgbSpace
 {
-    fn clone(&self) -> Hsl<Wp, T> { *self }
+    fn clone(&self) -> Hsl<S, T> { *self }
 }
 
 
-impl<T> Hsl<D65, T>
+impl<T> Hsl<Srgb, T>
     where T: Float,
 {
-    ///Linear HSL with white point D65.
-    pub fn new(hue: RgbHue<T>, saturation: T, lightness: T) -> Hsl<D65,T> {
+    ///HSL for linear sRGB.
+    pub fn new(hue: RgbHue<T>, saturation: T, lightness: T) -> Hsl<Srgb, T> {
         Hsl {
             hue: hue,
             saturation: saturation,
             lightness: lightness,
-            white_point: PhantomData,
+            space: PhantomData,
         }
     }
 }
 
-impl<Wp, T> Hsl<Wp, T>
+impl<S, T> Hsl<S, T>
     where T: Float,
-        Wp: WhitePoint<T>
+        S: RgbSpace
 {
     ///Linear HSL.
-    pub fn with_wp(hue: RgbHue<T>, saturation: T, lightness: T) -> Hsl<Wp, T> {
+    pub fn with_wp(hue: RgbHue<T>, saturation: T, lightness: T) -> Hsl<S, T> {
         Hsl {
             hue: hue,
             saturation: saturation,
             lightness: lightness,
-            white_point: PhantomData,
+            space: PhantomData,
+        }
+    }
+
+    #[inline]
+    fn reinterpret_as<Sp: RgbSpace>(self) -> Hsl<Sp, T> {
```

Comment:
It's there to transfer the values from one type to the other. The thing is that `S` and `Sp` are considered different, even if they are the same types. This is similar to `transmute`, but for types, and I'm keeping it private for now, since it's a bit sketchy from a semantic perspective.